Sources/MapboxMaps/Annotations/AnnotationImagesManager.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+
+internal protocol AnnotationImagesManagerProtocol: AnyObject {
+    func addImage(_ image: UIImage, id: String, sdf: Bool, contentInsets: UIEdgeInsets)
+    func removeImage(_ imageName: String)
+    func register(imagesConsumer: AnnotationImagesConsumer)
+    func unregister(imagesConsumer: AnnotationImagesConsumer)
+}
+
+@objc internal protocol AnnotationImagesConsumer {
+    func isUsingStyleImage(_ imageName: String) -> Bool
+}
+
+extension PointAnnotation {
+    public struct Image: Hashable {
+        public var image: UIImage
+        public var name: String
+
+        public init(image: UIImage, name: String) {
+            self.image = image
+            self.name = name
+        }
+    }
+}
+
+extension PointAnnotationManager: AnnotationImagesConsumer {
+
+    func isUsingStyleImage(_ imageName: String) -> Bool {
+        allImages.contains(imageName)
+    }
+}
+
+internal final class AnnotationImagesManager: AnnotationImagesManagerProtocol {
```

Comment:
This component potentially can be useful in other use cases(e.g. `Puck2D`) and it doesn't look like there is anything annotation specific about it, could it be moved closer to Style.swift and renamed e.g. to `StyleImageManager`?